k2/python/csrc/torch/fsa.cu

```diff
@@ -111,11 +122,109 @@ static void PybindFsaUtil(py::module &m) {
 
   // the following methods are for debugging only
   m.def("_fsa_to_fsa_vec", &FsaToFsaVec, py::arg("fsa"));
+
   m.def("_get_fsa_vec_element", &GetFsaVecElement, py::arg("vec"),
         py::arg("i"));
-  m.def("_create_fsa_vec", [](std::vector<Fsa *> &fsas) -> FsaVec {
-    return CreateFsaVec(fsas.size(), fsas.data());
-  });
+
+  m.def(
+      "_create_fsa_vec",
+      [](std::vector<Fsa *> &fsas) -> FsaVec {
+        return CreateFsaVec(fsas.size(), fsas.data());
+      },
+      py::arg("fsas"));
+
+  // returns Ragged<int32_t>
+  m.def("_get_state_batches", &GetStateBatches, py::arg("fsas"),
+        py::arg("transpose") = true);
+
+  m.def(
+      "_get_dest_states",
+      [](FsaVec &fsas, bool as_idx01) -> torch::Tensor {
+        Array1<int32_t> ans = GetDestStates(fsas, as_idx01);
+        return ToTensor(ans);
+      },
+      py::arg("fsas"), py::arg("as_idx01"));
+
+  m.def(
+      "_get_incoming_arcs",
+      [](FsaVec &fsas, torch::Tensor dest_states) -> Ragged<int32_t> {
+        Array1<int32_t> dest_states_array = FromTensor<int32_t>(dest_states);
+        return GetIncomingArcs(fsas, dest_states_array);
+      },
+      py::arg("fsas"), py::arg("dest_states"));
+
+  // returns Ragged<int32_t>
+  m.def("_get_entering_arc_index_batches", &GetEnteringArcIndexBatches,
+        py::arg("fsas"), py::arg("incoming_arcs"), py::arg("state_batches"));
+
+  // returns Ragged<int32_t>
+  m.def("_get_leaving_arc_index_batches", &GetLeavingArcIndexBatches,
+        py::arg("fsas"), py::arg("state_batches"));
+}
+
+template <typename T>
+static void PybindGetForwardScores(py::module &m, const char *name) {
+  // Return a std::pair
+  //   - forward_scores, a torch::Tensor of dtype torch.float32 or torch.float64
+  //   (depending on T) containing the scores
+  //
+  //   - best_path_arc_indexes (optional)
```

Comment:
mm.. this was called entering_arcs.  I don't like the name best_path_arc_indexes; this is traceback information.

Reply:
`entering_arcs` contains arc indexes for every state in the FsaVec, 
while `best_path_arc_indexes` contains arc indexes ONLY for the states that are on the best path.

Reply:
Will change it.

Reply:
Fixed.